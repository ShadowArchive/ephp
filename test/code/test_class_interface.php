<?php

interface A {
    const FOO = 100;
    public function foo();
}

## ONLY IN PHP 7
#interface B implements A {
interface B {
    const BAR = 200;
    public function bar($text);
    public function alice();
}

class C implements A, B {
    public function foo() {
        print self::FOO . "\n";
    }

    public function bar($text) {
        print C::BAR . " $text\n";
    }

    public function alice() {} 
}

$c = new C;
$c->foo();
$c->bar("EUR");

class D implements A, B {}

