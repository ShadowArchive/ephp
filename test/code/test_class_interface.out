100
200 EUR

Fatal error: Class D contains 2 abstract methods and must therefore be declared abstract or implement the remaining methods (A::foo, B::bar, B::alice) in {{CWD}}/test/code/test_class_interface.php on line 32
